Sources/Benchmark/BenchmarkFormat.swift

```diff
@@ -0,0 +1,6 @@
+import ArgumentParser
+
+public enum BenchmarkFormat: String, ExpressibleByArgument {
+    case console
+    case json
```

Comment:
Silly idea: what about having some of these support specifying a file in addition to dumping to stdout. (This could make scripting some automated performance comparisons a bit easier.)

Concretely, a workflow I'm imaging:
 1. Have an idea for a performance improvement. Clone the repo, run something like `swift run -c release Benchmarks --baseline benchmark/baseline.file` which writes out a baseline level of performance for your machine.
 2. Hack up a potential performance optimization.
 3. Run `swift run -c release Benchmarks --compare benchmark/baseline.file` which will give you the benchmark results, as well as the performance delta vs the baseline (including determining if it's statistically significant).

Now, I don't expect all of this is made to work in this PR, but maybe thinking through how we might want all of this to fit together could be good.

One other thing that I think will be important: being able to register "custom" recorders. Specifically, we want this library to subsume the benchmark infrastructure in [swift-models](https://github.com/tensorflow/swift-models/tree/master/Benchmarks), but it will need to integrate with the surrounding benchmark measurement infrastructure. Similarly, big companies may have an ecosystem of performance monitoring that would be good to plug into.

Reply:
Writing to stdout has been working great for us in swift-models. I think it's quite a bit simpler and can be trivially written to a file using pipes etc.

Reply:
Yeah, you're probably right. I'm wondering if it might make sense for the progress updates to go to stderr, however. That would give you the appropriate control to see progress as the benchmarks run while simultaneously allowing the JSON output to be redirected to a file... WDYT?

Reply:
Non-essential progress going to stderr is an excellent idea.

Reply:
@shabalind @saeta What are your thoughts about adopting [`swift-log`](https://github.com/apple/swift-log/) project-wide? There were a few places where I think the concept of reporting would be more cleanly understood and implemented as logging (for example, `report(finishedRunning:suite:nanosTaken:)`).

For more context, I recently wrote about [Swift Logging on NSHipster](https://nshipster.com/swift-log/).

Reply:
+1 for `swift-log`!

Reply:
I suggest we delay `swift-log` changes until a follow-up PR to unblock JSON output being merged sooner.